From 9f451c4435df1042a65e825195568c998f815bb5 Mon Sep 17 00:00:00 2001
From: Gabriel Roman <gabrielroman@google.com>
Date: Fri, 12 Jun 2026 10:45:58 -0700
Subject: [PATCH 8/8] Replace DeviceAdminInfo.loadDescription() with
 loadDescriptionSafe()

The loadDescriptionSafe() truncates the description string to a safe
1024 characters length.

Bug: 492093013
Test: manual
Flag: EXEMPT BUGFIX
Cherrypick-From: https://googleplex-android-review.googlesource.com/q/commit:d6a67bf84f1be105f2bc58275c6f84bf63c44d42
Security vulnerability cherry pick for https://b.corp.google.com/issues/492093013
Topic: https://googleplex-android-review.git.corp.google.com/q/topic:%22b492093013%22
#b/492093013
Cherrypick-From: https://googleplex-android-review.googlesource.com/q/commit:15a38f4b958648c87e986cc1f5117fa59b9c29a5
Merged-In: I1d29452e150538bcc576ac2d856029729a2ba51d
Change-Id: I1d29452e150538bcc576ac2d856029729a2ba51d
---
 .../specialaccess/deviceadmin/DeviceAdminAdd.java           | 2 +-
 .../specialaccess/deviceadmin/DeviceAdminListItem.java      | 2 +-
 .../specialaccess/deviceadmin/DeviceAdminListItemTest.java  | 6 +++++-
 3 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/com/android/settings/applications/specialaccess/deviceadmin/DeviceAdminAdd.java b/src/com/android/settings/applications/specialaccess/deviceadmin/DeviceAdminAdd.java
index 1425122feab..f6bb3f06bfe 100644
--- a/src/com/android/settings/applications/specialaccess/deviceadmin/DeviceAdminAdd.java
+++ b/src/com/android/settings/applications/specialaccess/deviceadmin/DeviceAdminAdd.java
@@ -661,7 +661,7 @@ public class DeviceAdminAdd extends CollapsingToolbarBaseActivity {
         mAdminName.setText(mDeviceAdmin.loadLabel(getPackageManager()));
         try {
             mAdminDescription.setText(
-                    mDeviceAdmin.loadDescription(getPackageManager()));
+                    mDeviceAdmin.loadDescriptionSafe(getPackageManager()));
             mAdminDescription.setVisibility(View.VISIBLE);
         } catch (Resources.NotFoundException e) {
             mAdminDescription.setVisibility(View.GONE);
diff --git a/src/com/android/settings/applications/specialaccess/deviceadmin/DeviceAdminListItem.java b/src/com/android/settings/applications/specialaccess/deviceadmin/DeviceAdminListItem.java
index f7dff2e2d7b..99be343e918 100644
--- a/src/com/android/settings/applications/specialaccess/deviceadmin/DeviceAdminListItem.java
+++ b/src/com/android/settings/applications/specialaccess/deviceadmin/DeviceAdminListItem.java
@@ -46,7 +46,7 @@ class DeviceAdminListItem implements Comparable<DeviceAdminListItem> {
         final PackageManager pm = context.getPackageManager();
         mName = mInfo.loadLabel(pm);
         try {
-            mDescription = mInfo.loadDescription(pm);
+            mDescription = mInfo.loadDescriptionSafe(pm);
         } catch (Resources.NotFoundException exception) {
             Log.w(TAG, "Setting description to null because can't find resource: " + mKey);
         }
diff --git a/tests/robotests/src/com/android/settings/applications/specialaccess/deviceadmin/DeviceAdminListItemTest.java b/tests/robotests/src/com/android/settings/applications/specialaccess/deviceadmin/DeviceAdminListItemTest.java
index 7392fb1376a..846a019c501 100644
--- a/tests/robotests/src/com/android/settings/applications/specialaccess/deviceadmin/DeviceAdminListItemTest.java
+++ b/tests/robotests/src/com/android/settings/applications/specialaccess/deviceadmin/DeviceAdminListItemTest.java
@@ -19,6 +19,8 @@ package com.android.settings.applications.specialaccess.deviceadmin;
 import static com.google.common.truth.Truth.assertThat;
 
 import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 import android.app.admin.DeviceAdminInfo;
@@ -60,13 +62,15 @@ public class DeviceAdminListItemTest {
         when(mDeviceAdminInfo.getActivityInfo()).thenReturn(new ActivityInfo());
         mDeviceAdminInfo.getActivityInfo().applicationInfo = new ApplicationInfo();
         when(mDeviceAdminInfo.loadLabel(any(PackageManager.class))).thenReturn(label);
-        when(mDeviceAdminInfo.loadDescription(any(PackageManager.class))).thenReturn(description);
+        when(mDeviceAdminInfo.loadDescriptionSafe(any(PackageManager.class)))
+                .thenReturn(description);
         when(mDeviceAdminInfo.loadIcon(any(PackageManager.class)))
                 .thenReturn(new ColorDrawable(Color.BLUE));
         when(mDeviceAdminInfo.getComponent()).thenReturn(cn);
 
         DeviceAdminListItem item = new DeviceAdminListItem(mContext, mDeviceAdminInfo);
 
+        verify(mDeviceAdminInfo, never()).loadDescription(any(PackageManager.class));
         assertThat(item.getKey()).isEqualTo("0@" + cn.flattenToShortString());
         assertThat(item.getName()).isEqualTo(label);
         assertThat(item.getDescription()).isEqualTo(description);
-- 
2.55.0

